### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by Andy-Python-Programmer**
I was unable to do this fix as I was buzy. Now I have done them.
```python
def setup():
    size(400, 400)

def draw():
    background(0)
    
    example = http_get("https://example.com")
    print(example.text)

run()
```
I have made it a class so that we can do example.text or example.json (for json). If we would have checked each of the lines for example decode json if it startswith("{") and endswith("}") then it will be a very slow process.

### CODE REVIEW

File: p5/io/http_data.py [hunk 1]
```diff
@@ -1,29 +1,38 @@
-from urllib import request, parse
+"""
+Part of the io library and has functions that require **internet** connection.
```
Comment (nakul-shahdadpuri) [inline p5/io/http_data.py:2]: remove "and" from comments

File: p5/io/http_data.py [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
+from urllib import request
 import json
 import requests
 
-# synchronous
-
+# Synchronous
 
-def http_get(url):
+class http_get:
```
Comment (nakul-shahdadpuri) [inline p5/io/http_data.py:11]: make the changes in docs accordingly.

File: p5/io/http_data.py [hunk 3]
```diff
@@ -1,29 +1,38 @@
...
+            print("Invalid url. Connection Failed.")
+            exit(1)
+
+        data = resp.read().decode("UTF-8")
+
+        self.text = data
+        try:
+            self.json = json.loads(data)
+        except: 
+            print("The data provided did not return in a json format.")
```
Comment (nakul-shahdadpuri) [inline p5/io/http_data.py:33]: return the data in string if not JSON.

**Review by nakul-shahdadpuri — CHANGES_REQUESTED** [review]

**Comment by parsoyaarihant** [comment]
can you be a bit more descriptive about this PR? 

It would be great if you could refer to the related issues and PRs.

**Comment by Andy-Python-Programmer** [comment]
@parsoyaarihant So the issue was that the get request from now only returned json so it failed it the result is not in json format. This pr fixes that issue. No related issues are there.

**Comment by parsoyaarihant** [comment]
I have few comments about this module: 

1. If the request failed, it should be more descriptive about why the request failed. it should give the status codes: https://www.restapitutorial.com/httpstatuscodes.html
2. If you are making a class based, implementation, all the functions should use that implementation. Not just http_get(). 
3. What about asynchronous calls? Should we pass a callback function for asynchronous requests? : https://p5js.org/reference/#/p5/httpGet

**Comment by Andy-Python-Programmer** [comment]
@parsoyaarihant for the 2. Point. I think we only need class for http_get we do not need class based implementation for http_post. And I will take a look at the http error codes.

**Comment by Andy-Python-Programmer** [comment]
Hi @parsoyaarihant about the error codes what we can do is just simply remove the try and except as URL lib will auto throw an exception and it will contain the respected http error codes.

File: p5/io/http_data.py [hunk 4]
```diff
@@ -1,29 +1,38 @@
...
+        except:
+            print("Invalid url. Connection Failed.")
+            exit(1)
+
+        data = resp.read().decode("UTF-8")
+
+        self.text = data
+        try:
+            self.json = json.loads(data)
+        except:
```
Comment (parsoyaarihant) [inline p5/io/http_data.py:32]: When do we expect `json.loads()` to fail?

**Comment by parsoyaarihant** [comment]
We can do that, I think the documentation and examples should also contain the try and except statements to explain how to use this API

File: p5/io/http_data.py (hunk 4 above)
Comment (Andy-Python-Programmer) [reply p5/io/http_data.py:32]: Well if the resp is 
```dart
test
```
then fail or 
```json
{wrong syntax
```

**Comment by parsoyaarihant** [comment]
Thanks

**Pull request merged**